odoo_updates/odoo_updates.py

```diff
@@ -171,6 +204,99 @@ def compare_translations(original_translations, modified_translations):
     return res
 
 
+def compare_fields(original_fields, modified_fields):
+    """
+    compares the fields of tables in a database and returns the direfencias
+    :param original_fields: This would be the fields from the point of
+     view of production database
+    :modified_fields: This are the changes made in the fields of
+    the database, the argument after the -u (-u all, -u app_module).
+    :return: a dict with the added, updated and deleted fields.
+    In the case of updated will return the diff between the org_database
+    and dst_database
+    """
+    def create_structure_model(records):
+        struct = {}
+        for rcd in records:
+            struct[rcd['model']] = []
+        for rcd in records:
+            struct[rcd['model']].append(rcd['name'])
+        return struct
+
+    def group(res, model, name, state, record):
+        values = res.get(state, [])
+        order = list(set([index for index,
+                     field in enumerate(values)
+                     for key, value in field.iteritems()
+                     if model == field['model'] and name == field['name']
+                     and state != 'updated' or model == field['model']
+                     and name == field['name']
+                     and isinstance(value, dict)]))
+        if order:
+            if state == 'updated':
+                res.get('updated')[order[0]].get('original', {})\
+                   .update(record['original'])
+                res.get('updated')[order[0]].get('modified', {})\
+                   .update(record['modified'])
+            else:
+                res.get(state)[order[0]].update(record)
+        else:
+            res.get(state).append(record)
+    res = {
+        'updated': list(), 'added': list(), 'deleted': list(),
+    }
+    checked = {'original': create_structure_model(original_fields),
+               'modified': create_structure_model(modified_fields)}
+    for modified in modified_fields:
+        for original in original_fields:
+            if modified['model'] == original['model']:
+                for column, values in modified.iteritems():
+                    if values != original[column] and modified['name']\
+                            == original['name']:
+                            updated = {
+                                'model': original['model'],
+                                'name': original['name'],
+                                'original': {column: original[column]},
+                                'modified': {column: values},
+                                }
+                            group(res, original['model'], original['name'],
+                                  'updated', updated)
```

Comment:
@mapuerta Also, why do you need this group function? just append the dictionaries to the respective list (updated, deleted or added) and place those lists in the res dictionary. Take a look at the compare_translations, compare_views and get_menus_diff and see how we sort the results, no need for extra functions.